app/src/main/resources/db/migration/V2__add_service_groups.sql

```diff
@@ -1,11 +1,23 @@
-
 CREATE TABLE IF NOT EXISTS service_groups (
     id BIGINT PRIMARY KEY AUTO_INCREMENT,
     name VARCHAR(255) NOT NULL,
     jurisdiction_id VARCHAR(255) NOT NULL,
     FOREIGN KEY (jurisdiction_id) REFERENCES jurisdictions(id)
 );
 
+-- create a default service group for each jurisdiction
+INSERT INTO service_groups (name, jurisdiction_id)
+    SELECT 'Default Service Group', id
+    FROM jurisdictions;
+
+ALTER TABLE services
+    ADD COLUMN service_group_id BIGINT;
+
+UPDATE services s
+    SET service_group_id = (SELECT id FROM service_groups sg WHERE sg.jurisdiction_id = s.jurisdiction_id AND sg.name = 'Default Service Group')
+    WHERE service_group_id IS NULL;
+
+
 ALTER TABLE services
-    ADD COLUMN service_group_id BIGINT NOT NULL,
+    MODIFY COLUMN service_group_id BIGINT NOT NULL,
     ADD FOREIGN KEY (service_group_id) REFERENCES service_groups(id);
```

Comment:
If we add a foreign key constraint to the services table without there being values for the `service_group_id` column it won't work (not when there is already data in the database at least). 

That's why we first add the `service_group_id` column and allow it to be null then we update the service table and ensure that every record has a value for the `service_group_id`, and only then do we make it a foreign key constraint.

Reply:
The idea is that we are creating the foreign key relationship between `service_groups` and `services` with the data first.  Then once the data is valid, we add the foreign key constraint.  If we add the constraint before we fix the data then it will cause an error.